src/util.ts

```diff
@@ -415,7 +415,9 @@ export class Util {
       try {
         parsedHttpRespBody.body = JSON.parse(body);
       } catch (err) {
-        parsedHttpRespBody.err = new ApiError('Cannot parse JSON response');
+        parsedHttpRespBody.err = new ApiError(
+          `Cannot parse response as JSON: ${body}`
+        );
```

Comment:
Since we're getting fancy, can we put a property on this ApiError instance called `body`?

Reply:
Are you opposed to having it in the message? Or do you want it to be in the `body` property in addition to the message?

I'd like to keep it right in the message. In my experience, only the error message, or the error message and its stack are logged and/or saved by high-level application error handlers for analysis after the fact. Using non-standard error properties makes it more likely that no one will get to see the body.

What do you think?

Reply:
Yes, definitely still in the message. Error messages are often eclipsed, and the user might want to format/display it without our prefix (`Cannot parse response as JSON:`).

Reply:
Okay. I see that `ApiError` already has a `response` property. How about adding `handleResp`'s `resp` to the error whenever there's an error, by adding the following to `handleResp`?

```ts
    if (parsedResp.err && resp) {
      parsedResp.err.response = resp
    }
```

If I understand correctly, that would make the body available through `err.response.body`.

Reply:
That seems like a good plan 👍

Reply:
Okay, I gave that a try in the latest commits.